docs/src/user-guide/guides-using-object-storage.md

```diff
@@ -0,0 +1,97 @@
+# Using object storage
+
+CLP can both compress logs from object storage (e.g., S3) and store archives on object storage. This
+guide explains how to configure CLP for both use cases.
+
+:::{note}
+Currently, only the [clp-json][release-choices] release supports object storage. Support for
+clp-text will be added in a future release.
+:::
+
+:::{note}
+Currently, CLP only supports using S3 as object storage. Support for other object storage services
+will be added in a future release.
+:::
+
+## Compressing logs from object storage
+
+To compress logs from S3, use the `s3` subcommand of the `compress.sh` script:
+
+```bash
+sbin/compress.sh s3 s3://<bucket-name>/<path-prefix>
+```
+
+* `<bucket-name>` is the name of the S3 bucket containing your logs.
+* `<path-prefix>` is the path prefix of all logs you wish to compress.
+
+:::{note}
+The `s3` subcommand only supports a single URL but will compress any logs that have the given path
+prefix.
+
+If you wish to compress a single log file, specify the entire path to the log file. However, if that
+log file's path is a prefix of another log file's path, then both log files will be compressed. This
+limitation will be addressed in a future release.
+:::
+
+## Storing archives on object storage
+
+To configure CLP to store archives on S3, update the `archive_output.storage` key in
+`<package>/etc/clp-config.yml`:
+
+```yaml
+archive_output:
+  storage:
+    type: "s3"
+    staging_directory: "var/data/staged-archives"  # Or a path of your choosing
+    s3_config:
+      region: "<aws-region-code>"
+      bucket: "<s3-bucket-name>"
+      key-prefix: "<s3-key-prefix>"
+      credentials:
```

Comment:
Should we note that we only support long term credential, as documented here https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html